### DISCUSSION TITLE
How do form attributes get used?

### DISCUSSION THREAD

**Discussion by amano-kenji** [jtdaugherty/brick#464] [Q&A] [answered]
```
, appAttrMap = \_ -> attrMap defAttr
    [ (L.listSelectedAttr, V.black `on` V.white)
    , (focusedFormInputAttr, V.black `on` V.yellow)
    , (invalidFormInputAttr, V.white `on` V.red) ]
```
I have this, but form inputs rendered by `renderForm` are displayed as white text in default background.

I use brick-1.7. Did I do something wrong? Or, is it a bug?

**Comment by jtdaugherty** [comment, marked as answer]
The behavior will depend on the particular form fields you are using. `focusedFormInputAttr` is only going to have an effect for form fields that don't use their own attributes already, like `List` or `Editor`. For an example of what I mean, try running `brick-form-demo`. You'll notice that it sets the same attributes that you do, but the form focused input attribute only affects the radio buttons. The editor attribute is used for the editors. This might be a bit confusing, but the only other option is to force the form input attributes to override the per-widget-type attributes, which might be even more undesirable. If this comment doesn't clear things up, please post a screenshot and we can explore further.

**Reply by amano-kenji** [reply]
I used `editTextField`. I think the specific attributes used by form fields should be documented.

**Reply by jtdaugherty** [reply]
Done in https://github.com/jtdaugherty/brick/commit/ea04c606810d6f702771d7b45709899044a767aa.

**Reply by amano-kenji** [reply]
:cat:
